ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewState.kt

```diff
@@ -1,17 +1,12 @@
 package com.revenuecat.purchases.ui.revenuecatui.data
 
-import com.revenuecat.purchases.Package
-import com.revenuecat.purchases.paywalls.PaywallData
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.TemplateConfiguration
 
 internal sealed class PaywallViewState {
     object Loading : PaywallViewState()
     data class Error(val errorMessage: String) : PaywallViewState()
-    data class Template2(
-        val paywallData: PaywallData,
-        val packages: List<Package>,
-        val selectedPackage: Package,
-        val shouldShowRestorePurchasesButton: Boolean,
-        val shouldShowTermsButton: Boolean,
-        val shouldShowPrivacyPolicyButton: Boolean,
+    data class Loaded(
+        val templateConfiguration: TemplateConfiguration,
+        val selectedPackage: TemplateConfiguration.PackageInfo,
```

Comment:
This state really only belongs to the `Paywall2` but still need to think about whether we want to have different states per template... I think that might be the easiest solution, even if we have to pass the `TemplateConfiguration` to each of the templates. Lmk if you were thinking of a different architecture @NachoSoto

Reply:
FWIW we're thinking of making every template support N packges. So definitely fine to have it here even if we ignore it in some templates for now.